Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -28,27 +32,57 @@ absolute values on a number field up to equivalence. (TODO)
 ring_norm, ostrowski
 -/
 
+/- ## Preliminary lemmas on limits -/
+
+open Filter
+
+-- If `a : ℝ` is bounded by `g(k)` for every `0 < k` then it is bounded by `lim_{k → ∞} g(k)`
+private lemma le_of_limit_le {a : ℝ} {g : ℕ → ℝ} {l : ℝ} (ha : ∀ (k : ℕ) (_ : 0 < k), a ≤ g k)
+  (hg : Tendsto g Filter.atTop (nhds l)) : a ≤ l := by
+  apply le_of_tendsto_of_tendsto tendsto_const_nhds hg
+  rw [EventuallyLE, eventually_atTop]
+  exact ⟨1, ha⟩
+
+
+-- For any `C > 0`, the limit of `C ^ (1/k)` is 1 as `k → ∞`
+private lemma tendsto_root_atTop_nhds_one {C : ℝ} (hC : 0 < C) : Tendsto
+    (fun k : ℕ ↦ C ^ (k : ℝ)⁻¹) atTop (nhds 1) := by
+  convert_to Tendsto ((fun k ↦ C ^ k) ∘ (fun k : ℝ ↦ k⁻¹) ∘ (Nat.cast))
+    atTop (nhds 1)
+  exact Tendsto.comp (Continuous.tendsto' (continuous_iff_continuousAt.2
+    (fun a ↦ Real.continuousAt_const_rpow (Ne.symm (ne_of_lt hC)))) 0 1 (Real.rpow_zero C))
+    <| Tendsto.comp tendsto_inv_atTop_zero tendsto_natCast_atTop_atTop
+
+--extends the lemma `tendsto_rpow_div` when the function has natural input
+private lemma tendsto_nat_rpow_div : Tendsto (fun k : ℕ ↦ (k : ℝ) ^ (k : ℝ)⁻¹)
+    atTop (nhds 1) := by
+  convert_to Tendsto ((fun k : ℝ ↦ k ^ k⁻¹) ∘ (Nat.cast) ) atTop (nhds 1)
+  apply Tendsto.comp _ tendsto_natCast_atTop_atTop
+  simp_rw [← one_div]
+  exact tendsto_rpow_div
+
+
 namespace Rat.MulRingNorm
 open Int
 
 variable {f g : MulRingNorm ℚ}
 
 /-- Values of a multiplicative norm of the rationals coincide on ℕ if and only if they coincide
-on `ℤ`. -/
+on `ℤ`. --/
```

Comment:
Please revert all these

Reply:
Done!

Reply:
Looks like you didn't actually revert these